### PR TITLE
Abort suid install if not root

### LINKED ISSUE #4227: INSTALL SUID should check that effective user is actually root
## Version of Singularity:

3.3.0

### Expected behavior

If mistakenly using `make install` instead of `sudo make install` and setuid install is enabled, fail with a diagnostic.

### Actual behavior

It installs the setuid binary setuid to my build user, which is worse than useless.

### Steps to reproduce behavior

omit sudo

### PR DESCRIPTION
**Pull request by sashayakovtseva**
**Description of the Pull Request (PR):**

This PR adds a small effective uid check during starter installation. If euid is not 0 then this installation doesn't make sense. It is better to abort it then to install with wrong suid bit.

**This fixes or addresses the following GitHub issues:** 

- Fixes #4227

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: mlocal/frags/build_runtime_suid.mk [hunk 1]
```diff
@@ -11,6 +11,10 @@ $(starter_suid): $(BUILDDIR)/.clean-starter $(singularity_build_config) $(starte
 		-o $@ $(SOURCEDIR)/cmd/starter/main_linux.go
 
 $(starter_suid_INSTALL): $(starter_suid)
+	@if [ `id -u` -ne 0 ] ; then \
+		echo "SUID is useful when running as root only, aborting"; \
```
Comment (cclerget) [inline mlocal/frags/build_runtime_suid.mk]:
```suggestion
		echo "SUID binary requires to execute make install as root, use sudo make install to finish installation"; \
```

**Review by cclerget — APPROVED** [review]
Small suggestion, feel free to apply otherwise LGTM

**Review by pisarukv — APPROVED** [review]

**Comment by jmstover** [comment]
Travis is failing with:
```
SUID binary requires to execute make install as root, use sudo make install to finish installation
```
If you do just a `make install` as a normal user, it also runs `install_SUID` doesn't it? You need the check in there for `with_suid -eq 1` don't you?

I don't know if this will fix Travis though... looks like it's building an RPM, as a normal user. When the RPM is installed by root, the permissions get applied. I _think_ permissions are separate. You can build a RPM as a normal user, it just tags what the permissions _should_ be.

So... attempting this could break that behavior... :/

**Review by jmstover — CHANGES_REQUESTED** [review]
Travis failing ... `with_suid -eq 1` check needed as well?

**Comment by sashayakovtseva** [comment]
@jmstover
> If you do just a make install as a normal user, it also runs install_SUID doesn't it?
It will run it if `with_suid` was true during `./mconfig` run.

 We don't need to check `with_suid` in makefile fragment b/c if that fragment appears in the resulting Makefile that means `with_suid` *was* set to true during `./mconfig` run.

**Comment by jmstover** [comment]
> We don't need to check `with_suid` in makefile fragment b/c if that fragment appears in the resulting Makefile that means `with_suid` _was_ set to true during `./mconfig` run.

`with_suid=1` by default
`with_suid=0` if `--without-suid` is set

**Review by jmstover — APPROVED** [review]

**Comment by jscook2345** [comment]
I could see some existing scripts breaking that rely on the current behavior, but I'm willing to merge this and see what our users come back with.

Thanks for this fix @sashayakovtseva !

**Pull request merged**